File: frontend/lib/legal.ts
export const TERMS_VERSION = "2026-07-28";
export const PRIVACY_VERSION = "2026-07-28";

export interface LegalConsentPayload {
  accepted_terms: true;
  terms_version: string;
  privacy_version: string;
}

export function currentLegalConsent(): LegalConsentPayload {
  return {
    accepted_terms: true,
    terms_version: TERMS_VERSION,
    privacy_version: PRIVACY_VERSION,
  };
}

File: backend/app/core/legal.py
"""Current legal-document versions used by account registration."""

from __future__ import annotations

TERMS_VERSION = "2026-07-28"
PRIVACY_VERSION = "2026-07-28"


def is_current_consent(
    accepted_terms: bool,
    terms_version: str,
    privacy_version: str,
) -> bool:
    """Return whether a registration accepted the currently published text."""
    return (
        accepted_terms
        and terms_version == TERMS_VERSION
        and privacy_version == PRIVACY_VERSION
    )
